### PR TITLE
add border replicate note

### LINKED ISSUE #7407: Documentation of medianBlur does not mention borderType
##### System information (version)

- OpenCV => 2.4.9.1
- Operating System / Platform => xubuntu
- Compiler => no idea

##### Detailed description

The documentation for medianBlur (http://docs.opencv.org/2.4/modules/imgproc/doc/filtering.html?highlight=median#cv2.medianBlur) fails to mention what happens to the borders. I think this is important because a first glance through the docs led me to believe that it would use `BORDER_DEFAULT`. After noticing this is not the case, it took me a while to find the correct answer in the source. In the case I was interested in, it uses `BORDER_REPLICATE` (https://github.com/opencv/opencv/blob/eddd417d74ab1b899fbd327c8dccf7429d26583b/modules/imgproc/src/smooth.cpp#L3113)

Please update the docs to include this. I know it's a detail, but it's the kind of detail good documentation should tell.

##### Steps to reproduce

```.py
from numpy import *
import cv2

orig = arange(7*8).reshape((7,8)).astype(uint8)
cvmedian = cv2.medianBlur(orig, 7)

data_portion = orig[:4, :4].copy()
border_default = cv2.copyMakeBorder(data_portion,3,0,3,0,cv2.BORDER_DEFAULT)
assert(median(border_default) != cvmedian[0,0])

border_replicate = cv2.copyMakeBorder(data_portion,3,0,3,0,cv2.BORDER_REPLICATE)
assert(median(border_replicate) == cvmedian[0,0])
```

### PR DESCRIPTION
**Pull request by StevenPuttemans**
resolves #7407

### CODE REVIEW

**Comment by sovrasov** [comment]
👍

**Comment by StevenPuttemans** [comment]
Noticed that the brackets where not needed, because they show up in the docs itself. So removed them :)

**Comment by mshabunin** [comment]
I suggest using `@note` for consistency.

**Comment by StevenPuttemans** [comment]
@mshabunin done!

File: modules/imgproc/include/opencv2/imgproc.hpp [hunk 1]
```diff
@@ -1321,6 +1321,8 @@ The function smoothes an image using the median filter with the \f$\texttt{ksize
 \texttt{ksize}\f$ aperture. Each channel of a multi-channel image is processed independently.
 In-place operation is supported.
 
+@note The median filter uses BORDER_REPLICATE internally to cope with border pixels, see cv::BorderTypes
```
Comment (alalek) [inline modules/imgproc/include/opencv2/imgproc.hpp]: cope -> copy?

Comment (StevenPuttemans) [inline modules/imgproc/include/opencv2/imgproc.hpp]: No, its the word `to cope with`, which means `to handle` border pixels.

**Pull request merged**